File: Project-Euler/Stats/euler4/euler4.py
import sys
import math

min = int(sys.argv[1])
max = int(sys.argv[2])
def digitsInArray(n):
	arr = []
	while n > 0:
		d = math.floor(n / 10)
		k = n - d * 10
		n = d
		arr.append(k)
	b = list(reversed(arr))
	return b

def isPalindrome(array):
	i = 0
	j = len(array)-1
	isPal = True
	while i <= j:
		if array[i] == array[j]:
			i += 1
			j -= 1
		else:
			isPal = False
			return isPal
	return isPal

def checkNumber(min, max):
	maxNum = 0
	found = False
	cycleDone = False
	while not(found):
		for x in range(max, min, -1):
			for y in range(max, min, -1):
				k = x * y
				result = isPalindrome(digitsInArray(k))
				if result:
					if k > maxNum:
						maxNum = k
						found = True
		cycleDone = True
		if cycleDone and not(found):
			break
	return maxNum
	
print checkNumber(min, max)

File: Project-Euler/Stats/euler4/euler4.java
import java.util.*;

public class euler4
{
	public static void main(String[] args)
	{
		int min = Integer.parseInt(args[0]);
		int max = Integer.parseInt(args[1]);
		System.out.println(checkNumber(min, max));
	}
	
	private static List<Integer> digitsInArray(int n)
	{
		List<Integer> arr = new ArrayList<>();
		int d;
		int k;
		while(n > 0)
		{
			d = (int) Math.floor(n / 10);
			k = n - d * 10;
			n = d;
			arr.add(k);
		}
		Collections.reverse(arr);
		return arr;
	}
	
	private static boolean isPalindrome(List<Integer> array)
	{
		int i = 0;
		int j = array.size()-1;
		boolean isPal = true;
		while(i <= j)
		{
			if(array.get(i) == array.get(j))
			{
				i += 1;
				j -= 1;				
			}
			else
			{
				isPal = false;
				return isPal;
			}
		}
		return isPal;
	}
	
	private static int checkNumber(int min, int max)
	{
		int maxNum = 0;
		boolean found = false;
		boolean cycleDone = false;
		int k;
		boolean result;
		while(!found)
		{
			for(int x = max; x >= min; x--)
			{
				for(int y = max; y >= min; y--)
				{
					k = x * y;
					result = isPalindrome(digitsInArray(k));
					if(result && k > maxNum)
					{
						maxNum = k;
						found = true;
					}
				}
			}
			cycleDone = true;
			if(cycleDone && !found)
			{
				break;
			}
		}
		return maxNum;
	}
}
